### PR TITLE
Bugfix/no pass reason

### PR DESCRIPTION
**Pull request by ori-gold-px**
No pass_reason on page_requested activity is almost certainly due to s2s_error. Added information about s2s_error in line with spec.

### CODE REVIEW

File: src/PerimeterxHttpClient.php [hunk 1]
```diff
@@ -35,9 +35,8 @@ public function __construct($config)
      * @inheritdoc
      * @return string
      */
-    public function send($url, $method, $json, $headers, $timeout = 0, $connect_timeout = 0)
+    public function send($pxCtx, $url, $method, $json, $headers, $timeout = 0, $connect_timeout = 0)
```
Comment (chennakarpx) [inline src/PerimeterxHttpClient.php:38]: not so important, but I would put pxCtx later in the function parameters (at least after the important parameters such as url and method)

File: src/PerimeterxActivitiesClient.php [hunk 2]
```diff
@@ -132,6 +132,12 @@ public function sendPageRequestedActivity($pxCtx)
         $details['module_version'] = $this->pxConfig['sdk_name'];
         $details['http_version'] = $pxCtx->getHttpVersion();
         $details['pass_reason'] = $pxCtx->getPassReason();
+        if ($details['pass_reason'] === "s2s_error") {
```
Comment (shaul-badusa-px) [inline src/PerimeterxActivitiesClient.php:135]: Please take to a seperate function

File: src/PerimeterxS2SValidator.php [hunk 3]
```diff
@@ -6,11 +6,51 @@
 
 class PerimeterxS2SValidator extends PerimeterxRiskClient
 {
-
     const RISK_API_ENDPOINT = '/api/v3/risk';
 
-    private function sendRiskRequest()
-    {
+    public function verify() {
```
Comment (chennakarpx) [inline src/PerimeterxS2SValidator.php:11]: great separation of the functions

**Review by shaul-badusa-px — APPROVED** [review]

File: tests/PerimeterxS2SValidatorTest.php [hunk 4]
```diff
@@ -1,79 +1,162 @@
...
+        $this->assertEquals("s2s_error", $pxCtx->getPassReason());
+        $this->assertEquals($s2sErrorReason, $pxCtx->getS2SErrorReason());
+        if (!empty($s2sErrorMessage)) {
+            $this->assertContains($s2sErrorMessage, $pxCtx->getS2SErrorMessage());
+        }
+        $this->assertEquals($httpStatus, $pxCtx->getS2SErrorHttpStatus());
+        $this->assertEquals($httpMessage, $pxCtx->getS2SErrorHttpMessage());
     }
 
+    public function providerS2SErrorData() {
```
Comment (chennakarpx) [inline tests/PerimeterxS2SValidatorTest.php:92]: provideS2S (redundant r)

**Review by chennakarpx — APPROVED** [review]

File: src/PerimeterxActivitiesClient.php (hunk 2 above)
Comment (ori-gold-px) [reply src/PerimeterxActivitiesClient.php]: Done

File: src/PerimeterxHttpClient.php (hunk 1 above)
Comment (ori-gold-px) [reply src/PerimeterxHttpClient.php]: Done

File: src/PerimeterxS2SValidator.php (hunk 3 above)
Comment (ori-gold-px) [reply src/PerimeterxS2SValidator.php:11]: Thanks!

File: tests/PerimeterxS2SValidatorTest.php (hunk 4 above)
Comment (ori-gold-px) [reply tests/PerimeterxS2SValidatorTest.php]: Done

**Review by shaul-badusa-px — APPROVED** [review]

**Pull request merged**